src/librustc/mir/tcx.rs

```diff
@@ -195,7 +209,7 @@ impl<'tcx> Rvalue<'tcx> {
                     }
                 }
             }
-            &Rvalue::InlineAsm { .. } => None
+            Rvalue::InlineAsm { .. } => None
```

Comment:
Huh, this is pre-existing and not a problem with this PR per se, but I think that **every** rvalue should have a type. The cases where we see `None` here are, I think, basically bugs. For example, inline-asm should probably be a *statement*, if it has no result, and `AggregateKind::Array` should store the type as part of its variant, to handle the empty case. Is there a strong reason for it not to be this way?